skyportal/handlers/api/comment.py

```diff
@@ -320,13 +321,13 @@ def post(self, associated_resource_type, resource_id):
             for user_mentioned in users_mentioned_in_comment:
                 self.flow.push(user_mentioned.id, "skyportal/FETCH_NOTIFICATIONS", {})
 
-        if isinstance(comment, Comment):
+        if comment.obj.id:  # comment on object, or object related resources
```

Comment:
Can you explain this check? This appears to be non-null even for comments/annotations on spectra

Reply:
See above.